qemu/tests/qsd_backup_pull.py

```diff
@@ -0,0 +1,297 @@
+                            for job in job_utils.query_jobs(self.qsd)]
+
+    def generate_inc_files(self):
+        return list(map(self.generate_data_file, self.src_img_tags))
+
+    def add_target_data_disks(self, bktype='full'):
+        """Hot add target disk to VM with qmp monitor"""
+        for tag in self.params.objects("source_images"):
+            image_params = self.params.object_params(tag)
+            img = image_params['full_backup_image'] if bktype == 'full' else image_params['inc_backup_image']
+            disk = self.target_disk_define_by_params(self.params, img)
+            disk.hotplug(self.qsd)
+            self.trash.append(disk)
+
+    def do_incremental_backup(self):
+        extra_options = {"sync": "none",
+                         "disabled_bitmaps": self.disabled_bitmaps,
+                         "wait_job_complete": False}
+        backup_utils.blockdev_batch_backup(
+            self.qsd,
+            self.source_images,
+            self.inc_backups,
+            self.inc_backup_bitmaps,
+            **extra_options)
+        self.backup_jobs = [job['id']
+                            for job in job_utils.query_jobs(self.qsd)]
+
+    def restart_vm_with_backup_images(self):
+        """restart vm with back2 as its data disk"""
+        self.main_vm.destroy()
+        self.qsd.stop_daemon()
+        self.params['qsd_images_qsd1'] = self.params['client_image_inc']
+        add_vubp_into_boot(self.params['client_image_inc'], self.params)
+        self.start_qsd()
+        self.prepare_main_vm()
+        self.clone_vm = self.main_vm
+
+    def clean_images(self):
+        for img in self.trash:
+            try:
+                if hasattr(img, 'remove'):
+                    img.remove()
+                else:
+                    sp_admin.remove_volume(img)
+            except Exception as e:
+                self.test.log.warn(str(e))
+
+    def rebase_backup_image(self):
+        """rebase image back2 onto back1"""
+        for i, img_obj in enumerate(self.inc_backup_client_images):
+            target_img_obj = self.full_backup_client_images[i]
+            img_obj.base_image_filename = target_img_obj.image_filename
+            img_obj.base_format = target_img_obj.image_format
+            img_obj.base_tag = target_img_obj.tag
+            img_obj.rebase(img_obj.params)
+
+    def verify_data_files(self):
+        non_existed_files = {}
+        disks_info = {}
+
+        # The last file should not exist on back2
+        for i, data_img in enumerate(self.src_img_tags):
+            non_existed_files[data_img] = self.files_info[data_img].pop()
+            disks_info[data_img] = self.disks_info[data_img]
+
+        # Check md5sum for the first two files
+        super(QSDBackupPullMode, self).verify_data_files()
+
+        # Check the files should not exist on back2
+        session = self.clone_vm.wait_for_login()
+        try:
+            for tag, info in six.iteritems(disks_info):
```

Comment:
six.iteritems -> disks_info.items, python2 will not be considered

Reply:
Have updated it, thanks @zhencliu